testfiles/gh-issue-1131.lvt

```diff
@@ -0,0 +1,26 @@
+\documentclass{minimal}
+\input{pgf-regression-test}
+
+\RequirePackage{pgfkeys}
+
+\begin{document}
+
+\START
+
+\BEGINTEST{pgfkeys: prevent csname from implicitly defining /foo}
+\pgfkeysgetvalue{/foo}{\mycmd}
+\pgfkeysifdefined{/foo}{\TYPE{true}}{\TYPE{false}}
+\ENDTEST
+
+\BEGINTEST{pgfkeys: keys must be able to hold \relax}
+\pgfkeyslet{/bar}{\relax} % or \pgfkeys{/bar/.initial=\relax}
```

Comment:
I don't think letting a key to `\relax` should “undefine” it. For that it's better to let it to something that is actually undefined.

Reply:
@hmenke agreed that `\pgfkeyslet{/bar}{\myundefinedmacro}` is better, but `\let\foo\relax` will have the same result for the majority of tests (and for the tests that have been implemented by now in `pgfkeys`).

Reply:
`\pgfkeysifdefined` has been using `\ifcsname` since 2007. I've not actually changed `\pgfkeysifdefined` but just cleaned up the accidental `\relax`ing. Now that I have squashed all the commits this should also be clearer.